src/components/TreebankService/TreebankService.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import {
+  MessagingService,
+  ResponseMessage,
+  WindowIframeDestination as Destination,
+} from 'alpheios-messaging/dist/dev/alpheios-messaging';
+
+const config = {
+  name: 'treebank',
+  targetIframeID: 'string-not-used',
+  targetURL: 'string-not-used',
+};
+
+const error = (request, message) => ResponseMessage.Error(request, new Error(message));
+
+class TreebankService extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { redirectTo: null };
+
+    this.service = new MessagingService('treebank-service', new Destination(config));
+
+    this.messageHandler = this.messageHandler.bind(this);
+  }
+
+  componentDidMount() {
+    const { name } = config;
+
+    // TODO this line is causing the error (even with try { }):
+    // TypeError: Cannot set property 'origin' of undefined
+    this.service.registerReceiverCallback(name, this.messageHandler);
+  }
+
+  componentWillUnmount() {
+    // TODO is there any way to unregister in alpheios-messaging?
```

Comment:
Is there some way to do the opposite of `registerReceiverCallback`? If there are event listeners involved I think we should get rid of them when they are no longer necessary. Though if there are no event listeners or anything that needs to be unset, then we don't need to unmount.

Reply:
this is a good suggestion.  I will enter an issue on the messaging service repo to request this change.

Reply:
alpheios-project/alpheios-messaging#4